### ISSUE TITLE
mailto links not working on Brave, when they are working on Chromium

### ISSUE THREAD

**Issue by life777eternal** [brave/brave-browser#20555]
## Description 
Mailto links are failing to open on Brave. When they are working fine on Chromium. Version 96.0.4664.110 (Official Build, ungoogled-chromium) (64-bit) Also copying the email with right-click copies the // in front of the email also. (Chromium copies the email with the "`mailto://`..." so it's worse.)

## Steps to Reproduce

   1. Find a website that has mailto link(s) with or without something predefined for the body, or make one.
   2. Right-click on the page and hit "Inspect".
   3. Left Click on the mailto link on Brave.
   4. Look for a similar issue at the bottom of the inspect section, under console. 
   5. Open the same website on Chromium.
   6. Click on the mailto link
   7. Notice the difference. 
   

## Actual result:
Don't get any new email opened.

![2022-01-18 - Brave](https://user-images.githubusercontent.com/25623649/150060696-de7af8e9-b030-49f5-a884-1260b7708864.png)

## Expected result:
A new email should open on my email client, with the info from the mailto link.

## Reproduces how often: 
Always

## Brave version:

1.34.80 Chromium: 97.0.4692.71 (Official Build) (64-bit)
adefa7837d02a07a604c1e6eff0b3a09422ab88d-refs/branch-heads/4692@{#1247}
Windows 7 Service Pack 1 (Build 7601.23964)
(An optimized version of Windows 7 Ultimate.)

## Version/Channel Information:

- Can you reproduce this issue with the current release? Aye
- Can you reproduce this issue with the beta channel? Unknown
- Can you reproduce this issue with the nightly channel? Unknown

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? Naw
- Does the issue resolve itself when disabling Brave Rewards? Unknown, not using Brave rewards
- Is the issue reproducible on the latest version of G. Chrome? Naw 

## Miscellaneous Information:
This site [here](https://w3docs.com/snippets/html/how-to-create-mailto-links.html) doesn't mention the `//` in front of the email address, though it works on Chromium and G. Chrome. 

Thank you.

**Comment by rebron** [comment]
@btlechowski Can you check this one when you get a chance? Windows 7 specific issue?

**Comment by a-e-egorov** [comment]
I have the same problem on Ubuntu 20.04.

**Comment by btlechowski** [comment]
Reproduced on

Brave | 1.34.81 Chromium: 97.0.4692.99 (Official Build) (64-bit) 
-- | --
Revision | d740da257583289dbebd2eb37e8668928fac5ead-refs/branch-heads/4692@{#1461}
OS | Windows 7 Service Pack 1 (Build 7601.24544)

Used links from https://www.scottseverance.us/mailto.html

Brave:
![image](https://user-images.githubusercontent.com/34715963/152194961-ce1cc4c0-404c-4e44-a746-1f5181a0cc83.png)

Chrome 97.0.4692.99
![image](https://user-images.githubusercontent.com/34715963/152195530-c2a181ff-ff42-458b-88c1-e576d637c949.png)

**Comment by dannypernik** [comment]
+1. I read through about 5 closed but unresolved topics in the Brave community from over a year ago before arriving here. Perhaps you could prompt the user to define mail handler on mailto click if handler isn't present. Not sure if this would be the ideal behavior.

**Comment by XilinJia** [comment]
> I have the same problem on Ubuntu 20.04. After clicking on "mailto", Brave pops up the window "Open with xdg-open"? I click button "Open xdg-open" and nothing happens.

I am using Xubuntu 22.04, so basically the same.  After clicking "Open xdg-open", brave opens thunderbird only showing the title bar with a blank window.  This is when thunderbird is not opened.  If however thunderbird app has been opened, then the correct send mail window is open with "Open xdg-open".
Firefox on the other hand opens the correct thunderbird send mail window regardless.

**Comment by d0ugparker** [comment]
I stumbled upon this, regarding Brave browser and mailto: links. I'm hoping it can help everyone.  

There was a little, double diamond icon showing in Brave's address bar. Mousing over it, it reported a “service handler request.”
 
![image](https://user-images.githubusercontent.com/11258443/225727362-66fe800b-b41c-4aa8-bbd8-9a8285acbf6a.png)

Clicking it, I received a dialog box leading me to what looks like the solution of solutions. I haven't gone down the different rabbit holes of radio button permutations, but I think this clears most of the log jam.

![image](https://user-images.githubusercontent.com/11258443/225727489-9a642da2-5941-4cfc-8cd6-ee589975516f.png)

**Comment by d0ugparker** [comment]
On further investigation, I went back and found a solution. (My answer, above, is NOT a solution. By the time you read this, I may have deleted it, but there may be value in leaving the old, not-so-good answer there for thoroughness.)

I had been using Brave browser using a "guest profile." Anecdotally, when Brave's version was upgraded in the summer of 2022, it seems that guest profiles no longer supported extensions, and an extension is required to add the handler that provides support for mailto: links. The solution then?

1. Create an actual, Brave profile: Hamburger menu > Create a new profile
2. Add the Brave, Gmail, mailto extension: Hamburger menu > Extension, then search for “Gmail mailto”
3. It's (probably) strongly suggested: confirm in Windows 11 that Brave is set as the default “app” for MAILTO—this takes a bit:
    Settings > Apps > Default Apps > Brave > MAILTO (mine is in caps) > Brave > Brave > "Set default"

Step (3) probably needs to be set in each browser, because as we all know, “It's not a bug, it's a _feature!"_

Doug P. Henderson, NV, USA

**Comment by gioliotta** [comment]
> On further investigation, I went back and found a solution. (My answer, above, is NOT a solution. By the time you read this, I may have deleted it, but there may be value in leaving the old, not-so-good answer there for thoroughness.)
> 
> I had been using Brave browser using a "guest profile." Anecdotally, when Brave's version was upgraded in the summer of 2022, it seems that guest profiles no longer supported extensions, and an extension is required to add the handler that provides support for mailto: links. Guest profiles that had worked with the extension in the past would suddenly no longer work. The solution?
> 
> 1. Create an actual, Brave profile: Hamburger menu > Create a new profile
> 2. Add the Brave, Gmail, mailto extension: Hamburger menu > Extensions > Web Store, then search for “Gmail mailto”
> 3. It's (probably) strongly suggested: confirm in Windows 11 that Brave is set as the default “app” for MAILTO—this takes a bit:
>    Settings > Apps > Default Apps > Brave > MAILTO (mine is in caps) > Brave > Brave > "Set default"
> 
> Step (3) probably needs to be set in each browser, because as we all know, “It's not a bug, it's a _feature!"_
> 
> Mods, please note: I don't know how to close the issue, or how to alert any of you that I need someone to consider closing the issue. I hope this helps.
> 
> Regards,
> 
> Doug P. Henderson, NV, USA

Thanks, it worked correctly for me

**Comment by tkralphs** [comment]
In case this is helpful to anyone, I just ran into this issue. Creating a profile first and then setting up GMail as a handler within that profile did indeed seem to work. Even without a profile, however, I finally figured out that if I go to brave://setttings/handlers and set GMail as the **default** handler, it starts to work. Despite being the _only_ handler, Gmail was apparently not the _default_.
